### ISSUE TITLE
Official matterport support for Tensorflow 2.x

### ISSUE THREAD

**Issue by vkramdev** [matterport/Mask_RCNN#2425]
The repository is using 1.1x version of tensorflow and tf2 is released and is more widely used version of tf.Also tf2 seems to be becoming default for newer implementation with keras. Does matterport have any plans to upgrade the repo to tf2 from the current tf1.x based implementation.

Thanks

**Comment by kimile599** [comment]
I don't see any official upgrade yet from matterport. But i see somewhere on kaggle someone is rewriting it in tf2.

**Comment by omar16100** [comment]
@kimile599 Was looking for it and couldn't find. Can you please share the link?

**Comment by kimile599** [comment]
> @kimile599 Was looking for it and couldn't find. Can you please share the link?

Sorry man. I used to see it somewhere in kaggle. I don't save the link

**Comment by kimile599** [comment]
> @kimile599 Was looking for it and couldn't find. Can you please share the link?

https://github.com/ahmedfgad/Mask-RCNN-TF2
I happen to find this here. Not sure if it is working.
Hope this helps.

**Comment by omar16100** [comment]
@kimile599 Thank you so much for sharing!

**Comment by kimile599** [comment]
> @kimile599 Thank you so much for sharing!

I have tried whole night for making this running but still couldn't.
It seems the tf  and keras version matching issues.
This one might be helpful. Let me know if you make it running and share your tf and keras versions.
https://github.com/matterport/Mask_RCNN/issues/1930

**Comment by omar16100** [comment]
@kimile599 I have tried a few times and still wasn't able to get it running.

Planning to experiment in the coming days and will update if I can get it working.

**Comment by kiflowb777** [comment]
Please look at:
https://blog.paperspace.com/mask-r-cnn-tensorflow-2-0-keras/
https://www.immersivelimit.com/tutorials/mask-rcnn-for-windows-10-tensorflow-2-cuda-101
https://github.com/tomgross/Mask_RCNN/tree/tensorflow-2.0
https://github.com/leekunhee/Mask_RCNN/tree/tensorflow2.0
https://github.com/akTwelve/Mask_RCNN
https://github.com/ahmedfgad/Mask-RCNN-TF2

**Comment by kiflowb777** [comment]
https://ngc.nvidia.com/catalog/resources/nvidia:mask_r_cnn_for_tensorflow2/files?version=20.06.3

**Comment by Tolga-Karahan** [comment]
Anyone managed to run library with tf2? I'm trying to install requirements below in Kaggle, but versions couldn't be found:
tf==1.15.2
Keras==2.2.4
Python==3.6
